torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -292,7 +292,8 @@ std::vector<int64_t> ShardingUtil::GetShardShape(
 
 std::vector<std::vector<at::indexing::TensorIndex>>
 ShardingUtil::GetShardIndicesForDevices(
```

Comment:
Would this work for local devices only?

Reply:
This will work for any set of devices, including nonlocal.